File: Backend/src/main/java/com/notes/Backend/services/NoteService.java
package com.notes.Backend.services;

import com.notes.Backend.domain.CreateNoteRequest;
import com.notes.Backend.domain.UpdateNoteRequest;
import com.notes.Backend.domain.entities.Note;
import com.notes.Backend.domain.entities.User;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
public interface NoteService {
    Note getNote(UUID id);
    List<Note> getAllNotes(UUID categoryId, UUID tagId);
    List<Note> getArchivedNotes(User user);
    Note createNote(User user, CreateNoteRequest createNoteRequest);
    Note updateNote(UUID id, UpdateNoteRequest updateNoteRequest);

    void deleteNote(UUID id);
}

File: Frontend/src/app/services/note.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {LoginRequest, LoginResponse} from '../model/interface/user';
import {Observable} from 'rxjs';
import {CreateNoteRequest, Note, noteCreated, Tag} from '../model/interface/note';
import {AuthService} from './auth.service';

@Injectable({
  providedIn: 'root'
})
export class NoteService {

  constructor(private http: HttpClient) { }

  authService = inject(AuthService);

  createNote(noteData: CreateNoteRequest): Observable<Note> {
    const headers = this.authService.getAuthHeaders();
    return this.http.post<Note>(
      "http://localhost:8080/api/v1/notes" ,noteData, {headers});
  }

  getAllNotes(): Observable<Note[]> {
    return this.http.get<Note[]>(
      "http://localhost:8080/api/v1/notes"
    )
  }
  getArchivedNotes(): Observable<Note[]> {
    const headers = this.authService.getAuthHeaders();
    return this.http.get<Note[]>(
      "http://localhost:8080/api/v1/notes/archived", {headers});
  }

  deleteNote(id: string): Observable<void> {
    const headers = this.authService.getAuthHeaders();
    return this.http.delete<void>(
      `http://localhost:8080/api/v1/notes/${id}`, {headers}
    )
  }

}
